src/mpi_domain.F90

```diff
@@ -224,19 +273,17 @@ subroutine exchange_mpi
       call exchange_scalar(d)
       call exchange_scalar(p)
       call exchange_scalar(phi)
-      do i = 1, spn
-         call exchange_scalar(spc(i,:,:,:)) ! TODO: This is inefficient, but works for now
-      enddo
       call exchange_scalar(v1)
       call exchange_scalar(v2)
       call exchange_scalar(v3)
       call exchange_scalar(b1)
       call exchange_scalar(b2)
       call exchange_scalar(b3)
-
       call exchange_scalar(e)
       call exchange_scalar(eint)
 
+      if (spn > 0) call exchange_spc(spc)
+
```

Comment:
To be consistent with the logic in `allocations.f90`, should we use `compswitch > 0` instead of `spn > 0` ?

Reply:
Good idea.

@ryosuke-hirai it wasn't quite clear to me what `compswitch` does. I see that the `spc` array gets allocated if `compswitch>0`, but then most other places in the code, `compswitch>=2` is used, including for boundary conditions.

What would be the correct switch to use to determine if the `spc` array should be communicated to MPI neighbours?

Reply:
There is a mode for `compswitch==1`, which is composition fixed to the mass coordinate. It is only relevant for 1D spherical coordinate problems assuming no mixing. It uses a fixed array `mudata` which is the mean molecular weight as a function of mass coordinate. At each time step, it interpolates the mean molecular weight onto each cell, which doesn't require any advection calculations (see `meanmolweight` in `composition.f90`). So I don't use `spc` for `compswitch==1`. I don't remember why I allocate `spc` when `compswitch>0` (and not `compswitch>=2`), but it may be because at some point I was using the `read_mesa` subroutine to read in the composition into `spc` first, and then constructing the `mudata` array from `spc`.

For MPI communication, we probably only need it when `compswitch>=2`.

Reply:
Thanks @ryosuke-hirai! This PR is ready from my end - feel free to merge it if you're happy with the changes.